src/rust/cryptography-x509/src/certificate.rs

```diff
@@ -14,6 +27,52 @@ pub struct Certificate<'a> {
     pub signature: asn1::BitString<'a>,
 }
 
+impl Certificate<'_> {
+    /// Returns the certificate's issuer.
+    pub fn issuer(&self) -> &NameReadable<'_> {
+        self.tbs_cert.issuer.unwrap_read()
+    }
+
+    /// Returns the certificate's subject.
+    pub fn subject(&self) -> &NameReadable<'_> {
+        self.tbs_cert.subject.unwrap_read()
+    }
+
+    /// Returns whether the certificate is "self-issued", whether its
+    /// issuer and subject are the same.
+    pub fn is_self_issued(&self) -> bool {
+        self.issuer() == self.subject()
+    }
+
+    /// Returns an iterable container over the certificate's extension, or
+    /// an error if the extension set contains a duplicate extension.
+    pub fn extensions(&self) -> Result<Extensions<'_>, CertificateError> {
+        self.tbs_cert
+            .extensions()
+            .map_err(CertificateError::DuplicateExtension)
```

Comment:
If we want a real error type, it should be propagated into the TBS extensions method.